test/gateway/eip20_gateway/progress_revert_stake.js

```diff
@@ -0,0 +1,456 @@
+        ZeroBytes,
+        revertStakeParams.blockHeight,
+        revertStakeParams.rlpParentNodes,
+      ),
+      'Message hash must not be zero.',
+    );
+
+  });
+
+  it('should fail when storage proof zero', async function () {
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        revertStakeParams.messageHash,
+        revertStakeParams.blockHeight,
+        '0x',
+      ),
+      'RLP parent nodes must not be zero.',
+    );
+  });
+
+  it('should fail when storage proof is incorrect', async function () {
+
+    let blockHeight = new BN(revokedProofData.co_gateway.confirm_stake_intent.proof_data.block_number,16);
+    let storageRoot = revokedProofData.co_gateway.confirm_stake_intent.proof_data.storageHash;
+
+    await gateway.setStorageRoot(blockHeight, storageRoot);
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        revertStakeParams.messageHash,
+        blockHeight,
+        revokedProofData.co_gateway.confirm_stake_intent.proof_data.storageProof[0].serializedProof,
+      ),
+      'Merkle proof verification failed.',
+    );
+
+  });
+
+  it('should fail when storage proof is invalid', async function () {
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        revertStakeParams.messageHash,
+        revertStakeParams.blockHeight,
+        "0x1245",
+      ),
+      'VM Exception while processing transaction: revert',
+    );
+
+  });
+
+  it('should fail when storage root is not committed for given block height', async function () {
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        revertStakeParams.messageHash,
+        revertStakeParams.blockHeight,
+        revertStakeParams.rlpParentNodes,
+      ),
+      'Storage root must not be zero.',
+    );
+
+  });
+
+  it('should fail when message is undeclared', async function () {
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        web3.utils.sha3("dummy"),
+        revertStakeParams.blockHeight,
+        revertStakeParams.rlpParentNodes,
+      ),
+      'StakeIntentHash must not be zero.',
+    );
+
+  });
+
+  it('should fail when message is progressed', async function () {
+
+    await gateway.setOutboxStatus(
+      revertStakeParams.messageHash,
+      MessageStatusEnum.Progressed,
+    );
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        revertStakeParams.messageHash,
+        revertStakeParams.blockHeight,
+        revertStakeParams.rlpParentNodes,
+      ),
+      'Message on source must be DeclaredRevocation.',
+    );
+
+  });
+
+  it('should fail when message status of inbox at target is declared', async function () {
```

Comment:
This and other tests around status of inbox  seems duplicate of `should fail when storage proof is incorrect`